dlt/extract/decorators.py

```diff
@@ -736,3 +756,39 @@ def _curry() -> TBoundItems:
         return _curry
 
     return _wrap
+
+
+def parallelize(f: Callable[TResourceFunParams, Any]) -> Callable[TResourceFunParams, Any]:
+    @wraps(f)
+    def _wrap(*args: Any, **kwargs: Any) -> Any:  # TODO: Type correctly
+        gen = f(*args, **kwargs)
+        if inspect.isfunction(gen):
```

Comment:
if `gen` is a function, we'll call it when pipe iterator is constructed. do not start evaluating gen here! context:
- pipes are cloned before evaluation. so if gen is generator function you'll be able to evaluate it multiple times
- maybe someone will pass the source to a different thread for evaluation. we support parallel pipelines now (and this decorator is made to support that)
instead check if this is generator function and if not raise a nice exception. please look into exceptions.py and use proper base class (where you can pass function name and a few other so user knows what is going on)